### ISSUE TITLE
"skaffold deploy" should not run build before

### ISSUE THREAD

**Issue by ahmadnurus** [GoogleContainerTools/skaffold#1352]
After upgrading skaffold to v0.19.0 `skaffold deploy` run build before

### Expected behavior

when run `skaffold deploy` it should deploy without running build before. like v0.18.0

```bash
$ /usr/local/Cellar/skaffold/0.18.0/bin/skaffold deploy
Starting deploy...
kubectl client version: 1.12
deployment.apps/account-api unchanged
deployment.extensions/account-db configured
persistentvolumeclaim/account-db unchanged
secret/account-db unchanged
service/account-db unchanged
deployment.apps/account-srv unchanged
deployment.apps/auth-api unchanged
configmap/auth-db-health unchanged
deployment.apps/auth-db configured
secret/auth-db unchanged
service/auth-db unchanged
deployment.apps/auth-srv unchanged
configmap/email-db-health unchanged
deployment.apps/email-db configured
secret/email-db unchanged
service/email-db unchanged
deployment.apps/email-srv unchanged
Deploy complete in 34.528770819s
There is a new version (0.19.0) of skaffold available. Download it at https://storage.googleapis.com/skaffold/releases/latest/skaffold-darwin-amd64
```

### Actual behavior

In skaffold v0.19.0 `skaffold deploy` run build before deploy

```bash
$ /usr/local/Cellar/skaffold/0.19.0/bin/skaffold deploy
Starting build...
Building [gcr.io/go-bookpub/account-api]...
Pushing code to gs://go-bookpub_cloudbuild/source/go-bookpub-06139abd6a64ca389756f622a8e70482.tar.gz
```

### Information

- Skaffold version: v0.19.0
- Operating system: MacOS 10.14.1
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v1alpha5 # when using v0.19.0 I change it to skaffold/v1beta1
kind: Config
build:
  googleCloudBuild:
    projectId: go-bookpub
    diskSizeGb: 100
    machineType: "UNSPECIFIED"
    dockerImage: gcr.io/cloud-builders/docker
  artifacts:
  # ------------- account ------------- #
  - image: gcr.io/go-bookpub/account-api
    context: account/api/
  - image: gcr.io/go-bookpub/account-srv
    context: account/srv/
  # --------------- auth -------------- #
  - image: gcr.io/go-bookpub/auth-api
    context: auth/api/
  - image: gcr.io/go-bookpub/auth-srv
    context: auth/srv/
  # --------------- email -------------- #
  - image: gcr.io/go-bookpub/email-srv
    context: email/srv/
deploy:
  kubectl:
    manifests:
  # ------------- account ------------- #
    - account/api/deployment.yml
    - account/db/*.yml
    - account/srv/deployment.yml
  # --------------- auth -------------- #
    - auth/api/deployment.yml
    - auth/db/*.yml
    - auth/srv/deployment.yml
  # --------------- email -------------- #
    - email/db/*.yml
    - email/srv/deployment.yml
profiles:
- name: local
  build:
    local:
      push: false
      useDockerCLI: true
```

**Comment by ahmadnurus** [comment]
Any update?

**Comment by WoLfulus** [comment]
I'm having problems with this too. It breaks my pipeline.

- I use `skaffold build` to build and push images
- Test the images normally with `docker run...`
- Deploy charts (in my case). 

Deploy rebuilding the images (and pushing them again) invalidates my previous tests

**Comment by nkubala** [comment]
@ahmadnurus sorry for the radio silence here. this is something we've been discussing the best way to address for a few months now, and have plans to tackle this quarter. in it's current form `skaffold deploy` isn't super useful, so we're aiming to fix that by essentially restructuring to have `skaffold build` + `skaffold deploy` = `skaffold run`.

**Comment by tejal29** [comment]
hmm i can add this as another task to verify when fixing #922

**Comment by tejal29** [comment]
This is fixed as part of #922
